### ISSUE TITLE
Add README to c++ and java bazel extractors. [phab:D1963]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2413]
_Created by **nbeloglazovgoog** at 2017-09-30 05:53:58_:

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-10-02 15:11:30_:

This is unlikely to actually work without additional configuration disabling the sandbox.  I would expect the extractor to be used via an external Bazel dependency:

`io_kythe//kythe/cxx/extractor:cxx_extractor_bazel`

(Although, sadly, given Kythe's current setup requirements, this is unlikely to work out of the box).

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-10-02 15:11:30_:

Would suggest "cxx.kindex" as the extension

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2017-10-02 15:11:30_:

cxx

**Comment by kythe** [comment]
_**nbeloglazovgoog** wrote at 2017-10-03 03:43:49_:

  - Replace c++ with cxx in filename extension.

**Comment by kythe** [comment]
_**nbeloglazovgoog** commented inline (context lost during migration) at 2017-10-03 03:47:51_:

I tried that initially. The problem I faced is that I have to include all libraries from Kythe's WORKSPACE file in my bazel project's WORKSPACE file: https://docs.bazel.build/versions/master/external.html#transitive-dependencies So it ends up quite a lot of work. I couldn't make it work at the end.

And using this hack with using binaries from Kythe installation directory seemed much easier. Also it doesn't require cloning the whole Kythe repo locally.
